pkg/controllers/workgenerator/controller.go

```diff
@@ -100,40 +101,83 @@ func (r *Reconciler) Reconcile(ctx context.Context, req controllerruntime.Reques
+		return controllerruntime.Result{}, controller.NewAPIServerError(true, err)
+	}
+
 	// make sure that the resource binding obj has a finalizer
 	if err := r.ensureFinalizer(ctx, &resourceBinding); err != nil {
 		return controllerruntime.Result{}, err
 	}
 
 	workUpdated := false
+	overrideSucceeded := false
 	// list all the corresponding works
 	works, syncErr := r.listAllWorksAssociated(ctx, &resourceBinding)
 	if syncErr == nil {
 		// generate and apply the workUpdated works if we have all the works
-		workUpdated, syncErr = r.syncAllWork(ctx, &resourceBinding, works)
+		overrideSucceeded, workUpdated, syncErr = r.syncAllWork(ctx, &resourceBinding, works, cluster)
 	}
 
-	if syncErr != nil {
-		klog.ErrorS(syncErr, "Failed to sync all the works", "resourceBinding", bindingRef)
-		// TODO: remove the deprecated "resourceBound" condition when we switch to the new condition model
-		resourceBinding.SetConditions(metav1.Condition{
-			Status:             metav1.ConditionFalse,
-			Type:               string(fleetv1beta1.ResourceBindingBound),
-			Reason:             syncWorkFailedReason,
-			Message:            syncErr.Error(),
-			ObservedGeneration: resourceBinding.Generation,
-		})
+	if overrideSucceeded {
+		overrideReason := condition.OverriddenSucceededReason
+		if len(resourceBinding.Spec.ClusterResourceOverrideSnapshots) == 0 &&
+			len(resourceBinding.Spec.ResourceOverrideSnapshots) == 0 {
+			overrideReason = condition.OverrideNotSpecifiedReason
+		}
 		resourceBinding.SetConditions(metav1.Condition{
-			Status:             metav1.ConditionFalse,
-			Type:               string(fleetv1beta1.ResourceBindingWorkCreated),
-			Reason:             syncWorkFailedReason,
-			Message:            syncErr.Error(),
+			Status:             metav1.ConditionTrue,
+			Type:               string(fleetv1beta1.ResourceBindingOverridden),
+			Reason:             overrideReason,
+			Message:            "Successfully applied the override rules on the resources",
 			ObservedGeneration: resourceBinding.Generation,
 		})
+	}
+
+	if syncErr != nil {
+		klog.ErrorS(syncErr, "Failed to sync all the works", "resourceBinding", bindingRef)
+		errorMessage := syncErr.Error()
+		// unwrap will return nil if syncErr is not wrapped
+		// the wrapped error string format is "%w: %s" so that remove ": " from messages
+		if err := errors.Unwrap(syncErr); err != nil && len(err.Error()) > 2 {
+			errorMessage = errorMessage[len(err.Error())+2:]
+		}
+
+		if !overrideSucceeded {
+			resourceBinding.SetConditions(metav1.Condition{
+				Status:             metav1.ConditionFalse,
+				Type:               string(fleetv1beta1.ResourceBindingOverridden),
+				Reason:             condition.OverriddenFailedReason,
+				Message:            fmt.Sprintf("Failed to apply the override rules on the resources: %s", errorMessage),
+				ObservedGeneration: resourceBinding.Generation,
+			})
+		} else {
```

Comment:
shouldn't we return all the errors?

Reply:
if override fails, we will stop creating/updating the work. So the workSynchronized condition will be always false.